tmt/steps/execute/scripts/tmt-report-result

```diff
@@ -0,0 +1,126 @@
+#!/bin/bash
+set -o errexit -o pipefail -o noclobber -o nounset
+
+server=
+outputFile=
+disablePlugin=
+port=
+message=
+help=False
+
+die() { echo "$*" >&2; exit 2; }  # complain to STDERR and exit with error
+needs_arg() { if [ -z "$OPTARG" ]; then die "No arg for --$OPT option"; fi; }
+
+position=
+check_opt_args () {
+    local OPTIND
+    while getopts s:o:p:t:port:message:outputfile:help-: OPT; do
+	# support long options: https://stackoverflow.com/a/28466267/519360
+        if [ "$OPT" = "-" ]; then   # long option: reformulate OPT and OPTARG
+            OPT="${OPTARG%%=*}"       # extract long option name
+        fi
+        case "$OPT" in
+            server )         needs_arg; eval "server=\"\$$OPTIND\"" ; OPTIND=$((OPTIND+1)) ;;
+            s )              needs_arg; server=$OPTARG ;;
+            outputfile )     needs_arg; eval "outputFile=\"\$$OPTIND\"" ; OPTIND=$((OPTIND+1)) ;;
+            o )              needs_arg; outputFile="$OPTARG" ;;
+            p )              needs_arg; disablePlugin="$OPTARG" ;;
+            disable-plugin ) needs_arg; eval "disablePlugin=\"\$$OPTIND\"" ; OPTIND=$((OPTIND+1)) ;;
+            t )              needs_arg; message="$OPTARG" ;;
+            message )        needs_arg; eval "message=\"\$$OPTIND\"" ; OPTIND=$((OPTIND+1)) ;;
+            port )           needs_arg; eval "port=\"\$$OPTIND\"" ; OPTIND=$((OPTIND+1)) ;;
+            no-plugins )     disablePlugin=* ;;
+            help )           help=True ;;
+            ??* )            die "Illegal option --$OPT" ;;  # bad long option
+            ? )              exit 2 ;;  # bad short option (error reported via getopts)
+        esac
+    done
+    shift $((OPTIND-1)) # remove parsed options and args from $@ list
+    position=$((OPTIND-1))
+}
+# Options wrapped in quotes to ensure -t/--message argument is parsed as a string phrase
+# rather than just the first string until a whitespace is encountered.
+check_opt_args "$@"
+shift $position # remove parsed options and args from $@ list
+# Return help options when command issued with no options or arguments.
+if [ $# -lt 2 ] || [ $help == True ]; then
+    echo "Usage:"
+    echo "  rstrnt-report-result [OPTION?] TASK_PATH RESULT [SCORE]"
+    echo ""
+    echo "Report results to lab controller. if you don't specify --port or"
+    echo "the server url you must have RECIPE_URL and TASKID defined."
+    echo "If HARNESS_PREFIX is defined then the value of that must be"
+    echo "prefixed to RECIPE_URL and TASKID"
+    echo ""
+    echo "Help Options:"
+    echo "  -h, --help                      Show help options"
+    echo ""
+    echo "Application Options:"
+    echo "  --port=PORT                     restraintd port number (Service default: 8081)"
+    echo "  -s, --server=URL                Server to connect to"
```

Comment:
It seems that `--server=URL` form of the option does not work, `--server URL` works fine. Similar for `--port`. We should probably suggest the working form in the usage message or fix it. From the other point of view, if these options are ignored by `tmt`, should we, for now, rather raise an error when they are used? @lukaszachy, any thoughts here?

Reply:
IMO just ignore them - If I have test which uses this command I definitely don't want to edit command parameters (especially if I want to keep test working in Beaker as usual).

Their usage is harmless from my PoV.

Reply:
This help text is included in the bash script exactly as it was output from the restraint service. I'm aware that the help incorrectly includes a '=' character between the option and argument where a space character should instead reside, but I wanted to include the text as is in case there were some existing tests which were checking the text in the help menu.

Reply:
I would not expect tests to be checking the `--help` message. Instead, I'd say it might be confusing for users if the help message provides irrelevant information. Shouldn't we rather explicitly state that, as for now, only the `RESULT` is processed?